### PR TITLE
Switch to AA2 SDK

### PR DESCRIPTION
**Pull request by elaydis**

### CODE REVIEW

**Review by FabioTacke — CHANGES_REQUESTED** [review]
Great work, @elaydis! Can't wait to get this thing merged. 😅

File: app/src/device/kotlin/de/digitalService/useID/idCardInterface/EidInteractionManager.kt [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+                val redirectUrl = authResult.url?.let { url ->
+                    val uriBuilder = url.buildUpon()
+                    majorCode.let { uriBuilder.appendQueryParameter("ResultMajor", it) }
+                    minorCode?.let { uriBuilder.appendQueryParameter("ResultMinor", it) }
+                    result.reason?.let { uriBuilder.appendQueryParameter("ResultMessage", it) }
+                    uriBuilder.build().toString()
+                }
+
+                if (majorCode != "error") {
+                    redirectUrl?.let { _eidFlow.value = EidInteractionEvent.AuthenticationSucceededWithRedirect(it) }
```
[the remark this answers is not in the document]
Comment (elaydis) [reply app/src/device/kotlin/de/digitalService/useID/idCardInterface/EidInteractionManager.kt]: Let me know what you think about the change that I made, I rewrote it like 5 times and still don't love it, but not sure if there's a a real elegant way to deal with the result 🙃

[the remark this answers is not in the document]
Comment (FabioTacke) [reply app/src/device/kotlin/de/digitalService/useID/idCardInterface/EidInteractionManager.kt]: That's way better now! 😃🙏

**Review by FabioTacke — APPROVED** [review]
🥳

**Pull request merged**